vite.config.ts

```diff
@@ -1,12 +1,34 @@
-import path from 'path';
+import path from 'node:path';
 import react from '@vitejs/plugin-react';
 import { defineConfig } from 'vite';
+import { VitePWA } from 'vite-plugin-pwa';
 
 export default defineConfig({
   server: {
     port: 3000,
   },
-  plugins: [react()],
+  plugins: [react(), 
+    VitePWA({
+      workbox: {
+        runtimeCaching: [
+          {
+            urlPattern: /^https:\/\/avatars\.githubusercontent\.com\/.*/i,
+            handler: 'StaleWhileRevalidate',
+            options: {
+              cacheName: 'gh-avatar-cache',
+              expiration: {
+                purgeOnQuotaError: true,
+                maxAgeSeconds: 24 * 60 * 60, // 1 day
```

Comment:
Can we increase the maxAge to 7days?

In most of cases the avatar will not change even when we refetch the fresh data.

We can add FAQ later, to state that we intentionally cache the avatar for 7 days.